e2etests/orchestration/create-single-instance-test.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# Copyright (C) 2024 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+usage="usage: $0 -d build_id -t target"
+
+build_id=
+target=
+
+while getopts ":hd:t:" opt; do
+  case "${opt}" in
+    h)
+      usage
+      exit 0
+      ;;
+    d)
+      build_id="${OPTARG}"
+      ;;
+    t)
+      target="${OPTARG}"
+      ;;
+    \?)
+      echo "Invalid option: ${OPTARG}" >&2
+      usage
+      exit 1
+      ;;
+    :)
+      echo "Invalid option: ${OPTARG} requires an argument" >&2
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+# Wait for the service to be available 
+#
+# $1 - http port to listen on
+wait_for_service() {
+  local counter=0
+  local wait_secs=4
+  while [[ "$counter" -lt 3 ]]
+  do
+    sleep ${wait_secs} 
+    if curl -sS "http://0.0.0.0:${1}/_debug/statusz"; then
+      return 
+    else
+      wait_secs=$((wait_secs*2))
+    fi
+    counter=$((counter+1))
+  done
+
+  echo "timeout starting service"
+  exit 1
+}
+
+http_port="61$(($RANDOM%10))$(($RANDOM%10))$(($RANDOM%10))"
```

Comment:
in presubmits there should not be collisions as this is the only test requiring port numbers, we should have a more elaborated solution for allocating ports when having more tests down the line. My next step, would be to use Golang to write these tests as we should avoid having more complex logic written in bash.

Reply:
Could you leave a TODO comment here, unless you have a plan to do that right after this change? I think it may cause a mysterious error for debuggers if we forget.

Reply:
I'm working on the Golang rewrite where we are not going to use randomly picked ports, so this bash code is going away.

Reply:
Great! No worries then :)